docs/architecture/architecture.md

```diff
@@ -106,7 +109,39 @@ This transition has been delayed until after the release of Clear Containers 3.0
 due to regressions, as described in [runtime issue 407](https://github.com/clearcontainers/runtime/issues/407).
 Once support for features like hotplug are available in `q35`, the project will
 look to transition to this machine type.
- 
+
+### Assets
+
+Additional resources are required to create each QEMU virtual machine
+instance, specifically a guest kernel and a root filesystem image known as the
+"mini O/S".
+
+The locations of these resources are passed to
+QEMU by the runtime (see [Configuration](configuration)).
+
+#### Guest kernel
+
+The guest kernel is passed to the QEMU hypervisor and used to boot the virtual
+machine. It is highly optimised for kernel boot time and minimal memory
+footprint, providing only those services required by a container workload.
+
+#### Root filesystem image
+
+The root filesystem image, sometimes referred to as the "mini O/S", is a
+highly optimised container bootstrap system based on [Clear Linux](https://clearlinux.org/).
+It provides an extremely minimal environment and has a highly optimised boot
+path.
+
+The only services running in the context of the mini O/S are the init daemon
+(`systemd`) and the [Agent](#agent). The real workload the user wishes to run
+executes within a confined context which uses the container manager image as
+its root filesystem.
+
+For example, when `docker run -ti ubuntu sh` is run, the hypervisor will boot
+the mini-OS image using the guest kernel causing `systemd` to launch
+`cc-agent`. The agent will create a confined context where `/bin/sh` runs
```

Comment:
We could clarify that the `/bin/sh` is really the (not sure what the real name is) 'docker entry point' - it sounds just vaguely like we will always run /bin/sh for a workload, or Ubuntu, which is not really the case - we run what the container manager has told us to run via the json etc.

Reply:
Agreed. I've reworded that section. How does it look now?